src/Client/LocalConnection.cpp

```diff
@@ -356,7 +356,7 @@ bool LocalConnection::poll(size_t)
 
         try
         {
-            pollImpl();
+            while (pollImpl());
```

Comment:
Let's add logging about an intention to retry because of a failed timeout. Also we have timeout in pullBlock, because the assumption was that we might want to send progress/metrics in between poll attempts. Which is done in lines 344-355. So currently this while loop in this place makes timeout pointless, as we could just pass 0 instead of timeout - which will just block until block is ready or query finished - an the effect will be the same. So I'd suggest to include lines 344-355 inside the loop + logging about poll retry because of failed timeout as I mentioned before.